Assets/MixedRealityToolkit-SDK/Features/Input/Handlers/ControllerPoseSynchronizer.cs

```diff
@@ -0,0 +1,140 @@
+        private Handedness handedness = Handedness.Left;
+
+        /// <summary>
+        /// The handedness this controller should synchronize with.
+        /// </summary>
+        public Handedness Handedness
+        {
+            get { return handedness; }
+            set
+            {
+                handedness = value;
+                ResetControllerReference();
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("Disables child GameObjects when the controller source is lost.")]
+        private bool disableChildren = true;
+
+        /// <summary>
+        ///Disables child <see cref="GameObject"/>s the controller source is lost.
+        /// </summary>
+        public bool DisableChildren
+        {
+            get { return disableChildren; }
+            set { disableChildren = value; }
+        }
+
+        /// <summary>
+        /// Is the controller this Synchronizer is registered to currently tracked?
+        /// </summary>
+        public bool IsTracked { get; private set; } = false;
+
+        private TrackingState lastTrackingState = TrackingState.NotTracked;
+
+        private uint controllerInputSourceId = 0;
+
+        #region IMixedRealitySourcePoseHandler Implementation
+
+        /// <inheritdoc />
+        public virtual void OnSourceDetected(SourceStateEventData eventData)
+        {
+            SetControllerReference(eventData);
+        }
+
+        /// <inheritdoc />
+        public virtual void OnSourceLost(SourceStateEventData eventData)
+        {
+            if (eventData.Controller.ControllerHandedness == Handedness)
+            {
+                ResetControllerReference();
+            }
+        }
+
+        /// <inheritdoc />
+        public virtual void OnSourcePoseChanged(SourcePoseEventData eventData)
+        {
+            if (controllerInputSourceId == 0)
+            {
+                SetControllerReference(eventData);
+            }
+
+            if (eventData.Controller.InputSource.SourceId != controllerInputSourceId)
+            {
+                return;
+            }
+
+            if (eventData.TrackingState != lastTrackingState)
+            {
+                IsTracked = eventData.TrackingState == TrackingState.Tracked;
+                lastTrackingState = eventData.TrackingState;
+            }
+
+            if (lastTrackingState == TrackingState.Tracked)
+            {
+                transform.position = eventData.MixedRealityPose.Position;
+                transform.rotation = eventData.MixedRealityPose.Rotation;
+            }
+        }
+
+        #endregion IMixedRealitySourcePoseHandler Implementation
+
+        #region Monobehaviour Implementation
+
+        protected override void OnEnable()
+        {
+            // Subscribe to interaction events
+            base.OnEnable();
+
+            if (DisableChildren)
+            {
+                gameObject.SetChildrenActive(false);
```

Comment:
this is a bit confusing, why aren't we activating children on enable?

Reply:
In master/June, the children are set inactive at first, then a check is made to try to find a controller. If one is found, they're immediately made active again in the callback.

Here, they're set active after SourceDetected, but that misses the case where a controller was detected before the prefab is instantiated.

This and ControllerFinder seem to have been ported incompletely or edited after the fact feature-wise, so I'm going to do a re-port soon.

Reply:
Good call; I missed that.